File: Support/Templates/content/CslaLibrary-CSharp/EditableChildList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Company.CslaLibrary1
{
  [Serializable]
  public class EditableChildList : 
    BusinessListBase<EditableChildList, EditableChild>
  {
    [FetchChild]
    private void Fetch(object childData)
    {
      RaiseListChangedEvents = false;
      var dataPortal = ApplicationContext.GetRequiredService<IChildDataPortal<EditableChild>>();
      foreach (var child in (List<object>)childData)
        Add(dataPortal.FetchChild(child));
      RaiseListChangedEvents = true;
    }
  }
}

File: Support/Templates/content/CslaLibrary-VisualBasic/EditableChildList.vb
Imports Csla.Security

<Serializable()> _
Public Class EditableChildList
  Inherits BusinessListBase(Of EditableChildList, EditableChild)

  <FetchChild>
  Private Sub Fetch(ByVal childData As Object)
    RaiseListChangedEvents = False
    Dim dataPortal As IChildDataPortal(Of EditableChild) = ApplicationContext.GetRequiredService(Of IChildDataPortal(Of EditableChild))()
    For Each child As Object In CType(childData, IList(Of Object))
      Me.Add(dataPortal.FetchChild(child))
    Next
    RaiseListChangedEvents = True
  End Sub

End Class
